src/playlists.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+import SpotifyWebApi = require('spotify-web-api-node');
+
+export class PlaylistsProvider implements vscode.TreeDataProvider<PlaylistAndTracks> {
+
+  constructor(private spotify: SpotifyWebApi) {}
+
+  async getChildren(element?: PlaylistAndTracks): Promise<PlaylistAndTracks[] | undefined> {
```

Comment:
You can guarantee returning an array if you initialize it from the beginning. Might not matter here but removing the possibility of `undefined` and `null` always makes life easier.

```suggestion
  private children: PlaylistAndTracks[] = [];
  
  constructor(private spotify: SpotifyWebApi) {}

  async getChildren(element?: PlaylistAndTracks): Promise<PlaylistAndTracks[]> {
```

Reply:
lol no we can't, `children` is undefined for tracks lol